sklearn/preprocessing/_encoders.py

```diff
@@ -38,44 +38,65 @@ def _check_X(self, X):
         - convert list of strings to object dtype
         - check for missing values for object dtype data (check_array does
           not do that)
+        - return list of features (arrays)
 
         """
-        X_temp = check_array(X, dtype=None)
-        if not hasattr(X, 'dtype') and np.issubdtype(X_temp.dtype, np.str_):
-            X = check_array(X, dtype=np.object)
+
+        if not (hasattr(X, 'iloc') and getattr(X, 'ndim', 0) == 2):
+            X_temp = check_array(X, dtype=None)
+            if not hasattr(X, 'dtype') and\
+               np.issubdtype(X_temp.dtype, np.str_):
+                X = check_array(X, dtype=np.object)
+            else:
+                X = X_temp
+            needs_validation = False
         else:
-            X = X_temp
+            needs_validation = True
```

Comment:
and here a comment like `# pandas dataframe, do validation later column by column`